registrations.md

```diff
@@ -0,0 +1,52 @@
+
+If all L2 nodes are destroyed along with their enclave-derived key pairs, the shared secret key will be lost, and the confidential contract state can never again be processed. As long as one L2 node remains bootable and can recreate its enclave-derived key, it may decrypt the shared secret and share it with other nodes. The incentives must ensure that there is enough geographical distribution to make this scenario impossible.
+
+### Attestation Verification
+The solution described above assumes that attestation verification can be implemented efficiently as part of the Network Management contract. This is the ideal solution since it makes the contract the root of trust for the L2 network. The governance mechanism of the management contract will control the attestation requirements, like the hash of the program.
+
+### Aggregator Registration
+An Aggregator is a special type of L2 node which has the power to be a sequencer in some rounds and submit rollups.
+
+In addition to node registration, there is an additional step of pledging a stake.
+
+The sequence for node registration is shown in the following diagram:
+![aggregator staking](./images/aggregator-stake.png)
+
+These are the steps to become an aggregator.
+* Register with the L1 Network Management contract and pay a significant stake in the Obscuro token. The stake has multiple roles. The first one is to penalize aggregators who attempt to hack the protocol, and second is for the aggregators to buy into the ecosystem, so that they will make an effort to keep it running smoothly.
+* Set up a server with a valid, unaltered, up-to-date and secured TEE and provide an attestation from the hardware manufacturer or a delegate to the management contract.
+* On seeing this request to join the network published to the L1, another registered TEE will share the shared secret, used to encrypt and decrypt user transactions.
+* Once in possession of the secret, the TEE can start processing all the L2 transactions that are stored on the L1 blockchain and build the state.
+* Once this is completed, the new aggregator can join the gossip with the other aggregators and participate in the lottery for producing rollups.
+* Some end users will send encrypted instructions directly to this server, and it will gossip these with other nodes in the L2 network, encrypted with the shared secret.
+* As aggregators process messages, they maintain the L2 state in the encrypted TEE memory or encrypted in a local database. If they are the winner of the lottery, when the time comes they will create a valid rollup and publish it to L1.
+* All aggregators keep track of the blocks submitted to the management contract to make sure they are up-to-date with the source of truth.
+* The first aggregator to register has a special role, as it has to create the shared secret.
+
+Note: Each aggregator needs an ETH balance on the L1 to pay for the submission of the rollup.
+
+### User Registration
```

Comment:
I would put this in a separate chapter, now that I look at it.

Reply:
Here we mention "the Obscuro wallet". Should the text refer to an "Obscuro-enabled wallet"? We use this elsewhere in the WP.

Reply:
good point. let's stick with "Obscuro-enabled wallet"